src/Pagination/index.js

```diff
@@ -70,43 +71,57 @@ const Pagination = ({
   onPageChange = noop,
   totalPages = 1,
   defaultSelectedPage = 1,
+  selectedPage: selectedPageControlled,
   testId,
 }) => {
-  // no jokers allowed 🤡
-  // - default to first page if default selected page is out of bounds
-  // - enforce minimum of 1 pages
-  if (defaultSelectedPage > totalPages || defaultSelectedPage < 1) {
-    defaultSelectedPage = 1;
-  }
-  if (totalPages < 1) {
-    totalPages = 1;
-  }
-
+  const isControlled = selectedPageControlled !== undefined;
   const [selectedPage, setSelectedPage] = useState(defaultSelectedPage);
+  const [paginationAttributes, setPaginationAttributes] = useState(
+    _getAttributes(totalPages, selectedPage)
+  );
+
+  useEffect(() => {
+    const isOutOfBounds = totalPages < selectedPageControlled;
+    if (isControlled) {
+      setSelectedPage(isOutOfBounds ? 1 : selectedPageControlled);
+      setPaginationAttributes(
+        _getAttributes(totalPages, selectedPageControlled)
+      );
+    } else {
+      setPaginationAttributes(_getAttributes(totalPages, selectedPage));
+    }
+  }, [totalPages, selectedPageControlled, selectedPage]);
+
   const {
     visiblePages,
     selectedIndex,
     showPrev,
     showNext,
     firstPage,
     lastPage,
-  } = _getAttributes(totalPages, selectedPage);
+  } = paginationAttributes;
```

Comment:
This gets updated on first render, and any time `totalPages`, `selectedPageControlled`, or `selectedPage` (uncontrolled) changes